crates/bitwarden/src/vault/send.rs

```diff
@@ -200,16 +200,16 @@ impl KeyDecryptable<SymmetricCryptoKey, SendView> for Send {
         // For sends, we first decrypt the send key with the user key, and stretch it to it's full
         // size For the rest of the fields, we ignore the provided SymmetricCryptoKey and
         // the stretched key
-        let k: Vec<u8> = self.key.decrypt_with_key(key)?;
-        let key = Send::derive_shareable_key(&k)?;
+        let k: DecryptedVec = self.key.decrypt_with_key(key)?;
+        let key = Send::derive_shareable_key(k.expose())?;
 
         Ok(SendView {
             id: self.id,
             access_id: self.access_id.clone(),
 
             name: self.name.decrypt_with_key(&key).ok().unwrap_or_default(),
             notes: self.notes.decrypt_with_key(&key).ok().flatten(),
-            key: Some(URL_SAFE_NO_PAD.encode(k)),
+            key: Some(URL_SAFE_NO_PAD.encode(k.expose())),
```

Comment:
Should we expose some `map` function on Decrypted types to make sure that the b64 encoded value is also protected?

```rust
  pub fn map_ref<W: Zeroize>(self, func: fn(&V) -> W) -> Decrypted<W> {
      Decrypted::new(Box::new(func(&self.value)))
  }

key: Some(k.map_ref(|k| URL_SAFE_NO_PAD.encode(k))) // This returns a Decrypted<String>
```

Reply:
Maybe, we need to be a bit careful about the return value and ensure we zeroize it to our ability.